### PR TITLE
Close CPPL-218

### PR DESCRIPTION
**Pull request by sorbeee**

### CODE REVIEW

File: src/Client/View/MainPage/MainPage.cpp [hunk 1]
```diff
@@ -96,3 +96,9 @@ void MainPage::OnLogoutClicked()
 {
 	emit Logout();
 }
+
+void MainPage::ShowBalacne(const Balance& money)
```
Comment (iodream) [inline src/Client/View/MainPage/MainPage.cpp:100]:
```suggestion
void MainPage::ShowBalance(const Balance& money)
```

**Review by AndriyFedorenko — APPROVED** [review]

File: src/Client/Controller/MainPage/MainPageController.cpp [hunk 2]
```diff
@@ -175,3 +178,12 @@ void MainPageController::OnChangeSubPage(MainSubPages page, PageData data)
 {
 	ChangeSubPage(page, data);
 }
+
+Balance MainPageController::UpdateUserBalance(const IdType &id)
+{
+	GetBalanceModel model{m_hostname};
+	auto request = model.FormRequest(id);
+	auto response = m_http_client.Request(request);
```
Comment (iodream) [inline src/Client/Controller/MainPage/MainPageController.cpp:186]: please have some error checking from http_client here like in OnUpdateList for example

**Review by iodream — CHANGES_REQUESTED** [review]

**Review by VasylDobrovolskyi — APPROVED** [review]

File: src/Client/Controller/MainPage/MainPageController.cpp [hunk 3]
```diff
@@ -175,3 +178,20 @@ void MainPageController::OnChangeSubPage(MainSubPages page, PageData data)
 {
 	ChangeSubPage(page, data);
 }
+
+std::optional<Balance> MainPageController::UpdateUserBalance(const IdType &id)
+{
+	GetBalanceModel model{m_hostname};
+	auto request = model.FormRequest(id);
+	auto response = m_http_client.Request(request);
```
Comment (iodream) [inline src/Client/Controller/MainPage/MainPageController.cpp:186]: m_http_client.Request may throw an exception. 
add  a try-catch with
```
catch(Poco::Exception& exc)
	{
		return;
	}
```
here

File: src/Client/Controller/MainPage/MainPageController.cpp [hunk 4]
```diff
@@ -139,6 +141,8 @@ bool MainPageController::UpdateSubPage(MainSubPages page, PageData data)
 {
 	bool update_succeeded{true};
 
+	m_page.ShowBalacne(*UpdateUserBalance(m_user_id));
```
Comment (iodream) [inline src/Client/Controller/MainPage/MainPageController.cpp:144]: typo
```suggestion
	m_page.ShowBalance(*UpdateUserBalance(m_user_id));
```

**Review by iodream — CHANGES_REQUESTED** [review]

**Pull request merged**